var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -113,15 +136,37 @@ def die_without_fortran(self):
                 'Mpich requires both C and Fortran compilers!'
             )
 
+    def with_or_without_tcp(self, activated):
+        if activated:
+            return '--with-device=ch3:nemesis:tcp'
+
+    def with_or_without_mxm(self, activated):
+        if activated:
+            return '--with-device=ch3:nemesis:mxm'
+
+    def with_or_without_ucx(self, activated):
+        if activated:
+            return '--with-device=ch4:ucx'
+
+    def with_or_without_ofi(self, activated):
+        if activated:
+            if 'device=ch4' in self.spec:
+                return '--with-device=ch4:ofi'
+            else:
+                return '--with-device=ch3:nemesis:ofi'
+
     def configure_args(self):
         spec = self.spec
-        return [
+        config_args = [
             '--enable-shared',
             '--with-pm={0}'.format('hydra' if '+hydra' in spec else 'no'),
             '--with-pmi={0}'.format('yes' if '+pmi' in spec else 'no'),
             '--{0}-romio'.format('enable' if '+romio' in spec else 'disable'),
-            '--{0}-ibverbs'.format('with' if '+verbs' in spec else 'without')
+            '--{0}-ibverbs'.format('with' if '+verbs' in spec else 'without'),
         ]
+        config_args.extend(self.with_or_without('netmod'))
+
+        return filter(None, config_args)
```

Comment:
Is this a good idea? Local testing wouldn't work when there were `None` elements in the list. This was the best I could come up with given the quirky nature of our `--with-device=` flag.

Reply:
I think the Travis results answered my question for me...